### PR TITLE
Add USM metrics to agent telemetry

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

Adds agent telemetry that mirrors the metrics USM was already collecting for expvar.

### Motivation

Conversion to agent telemetry internally.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/protocols/kafka/telemetry.go [hunk 1]
```diff
@@ -13,9 +13,17 @@ import (
...
 
 	libtelemetry "github.com/DataDog/datadog-agent/pkg/network/protocols/telemetry"
+	pkgtelemetry "github.com/DataDog/datadog-agent/pkg/telemetry"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
+var telem = struct {
+	hits, dropped pkgtelemetry.Counter
+}{
+	pkgtelemetry.NewCounter("usm__kafka", "hits", []string{}, ""),
```
Comment (leeavital) [inline pkg/network/protocols/kafka/telemetry.go:23]: Is this emitted as `usm.kafka.hits` ?

**Review by leeavital — APPROVED** [review]

File: pkg/network/encoding/encoding.go [hunk 2]
```diff
@@ -29,6 +30,11 @@ var (
 
 	cfgOnce  = sync.Once{}
 	agentCfg *model.AgentConfiguration
+	stats    = struct {
+		orphanAggregations pkgtelemetry.Counter
+	}{
+		pkgtelemetry.NewCounter("usm", "orphan_aggregations", []string{"protocol"}, ""),
+	}
```
Comment (guyarb) [inline pkg/network/encoding/encoding.go:37]: why do you need the struct definition here?

Comment (brycekahle) [reply pkg/network/encoding/encoding.go:37]: Not required, but keeps it consistent with other uses of telemetry, and makes it easier to add more metrics in the future.

**Review by guyarb — APPROVED** [review]

File: pkg/network/protocols/http/statkeeper.go [hunk 3]
```diff
@@ -85,6 +85,7 @@ func (h *StatKeeper) Close() {
 func (h *StatKeeper) add(tx Transaction) {
 	rawPath, fullPath := tx.Path(h.buffer)
 	if rawPath == nil {
+		telem.malformed.Inc(h.telemetry.proto)
 		h.telemetry.malformed.Add(1)
```
Comment (guyarb) [inline pkg/network/protocols/http/statkeeper.go:89]: why do we need the duplication of telemetries?

Comment (guyarb) [reply pkg/network/protocols/http/statkeeper.go:89]: same for all other locations

**Review by guyarb — CHANGES_REQUESTED** [review]
CRed again

File: pkg/network/protocols/http/statkeeper.go (hunk 3 above)
Comment (brycekahle) [reply pkg/network/protocols/http/statkeeper.go:89]: As mentioned in the PR description, collect metrics via expvar is being deprecated internally, so anything that was only on expvar needs to be exposed via agent telemetry.

File: pkg/network/protocols/http/telemetry.go [hunk 4]
```diff
@@ -8,14 +8,29 @@
...
+	pkgtelemetry.NewCounter("usm__http", "dropped", []string{"protocol"}, ""),
+	pkgtelemetry.NewCounter("usm__http", "rejected", []string{"protocol"}, ""),
+	pkgtelemetry.NewCounter("usm__http", "malformed", []string{"protocol"}, ""),
+	pkgtelemetry.NewCounter("usm__http", "aggregations", []string{"protocol"}, ""),
+}
+
 type Telemetry struct {
+	proto                                       string
 	LastCheck                                   *atomic.Int64
 	hits1XX, hits2XX, hits3XX, hits4XX, hits5XX *libtelemetry.Metric
```
Comment (DanielLavie) [inline pkg/network/protocols/http/telemetry.go:35]: Why are these a part of the `libtelemetry` and the other metrics are `pkgtelemetry` counter telemetry?

**Review by guyarb — APPROVED** [review]

**Comment by p-lambert** [comment]
I ended up merging a [PR](https://github.com/DataDog/datadog-agent/pull/18151) fully deprecating expvar and adding support for Prometheus to our (USM) internal telemetry library, so I think we can close this PR

**Pull request closed without merging**